### ISSUE TITLE
Ember integration for apollo

### ISSUE THREAD

**Issue by boydkr** [apollostack/apollo#41]
I see there are integrations already for React and Angular.  I would be very interested in trying out apollo if it were available for Ember

**Comment by stubailo** [comment]
For an integration to happen we would need someone who is very interested in it to be the primary maintainer - I'm happy to help with design advice, getting started, etc but I don't personally use ember so I don't think I would be a good person to start an ember integration.

Would you be interested in working together on an ember library?

**Comment by jlevycpa** [comment]
@stubailo Have you heard of anyone working on this? I've been thinking about experimenting with an Ember integration but I don't want to duplicate effort.

**Comment by stubailo** [comment]
Haven't heard anything yet!

**Comment by jlevycpa** [comment]
@stubailo Here is a start. I'd love to get your thoughts on the approach. 
https://github.com/jlevycpa/ember-apollo

**Comment by stubailo** [comment]
This looks great! Mind sending a PR adding a page like this one? http://dev.apollodata.com/core/vue.html

Also, do you have an example of an app using this? I haven't used Ember before, but I'd be happy to give it a look.

**Comment by jlevycpa** [comment]
Thanks! This isn't ready for prime time yet, I just wanted to put what I had out there so people know it's in progress (and can help if they want). Do you want this linked to your docs now? Or not until its more stable?

No example app yet, that's my next step. It probably makes sense to do an Ember port of GitHunt, no?

Just out of curiosity, what's the distinction between the "featured" integrations like React and Angular (with full docs), and the others which are a bit hidden and just link out to GitHub READMEs? What would it take to get Ember up alongside React and Angular? Is it just a matter of building out the docs? Or are you looking for a certain level of interest in building apps using Apollo from the Ember community? Or vise versa? I'm asking because I think Ember is a really great way to build apps. I moved from Meteor to React/Webpack to Ember, and I think Ember has a lot to offer most app developers. It combines the awesome "batteries included" experience that I loved about Meteor with a powerful view layer that I think is superior to React in some ways, but without dictating how you handle the server side. I'm not a big fan of Ember Data, the (optional) client side data solution, but substitute in Apollo and I think you have a nearly ideal solution for many use cases.

However, I think Ember has a bit of a marketing/visibility problem, so once there is a feature complete, stable integration, I'm interested to see how I can promote the Ember/Apollo combination. Part of that is promoting Apollo within the Ember community, but part of that is making Ember visible as a choice to people interested in Apollo as well.

**Comment by stubailo** [comment]
> what's the distinction between the "featured" integrations like React and Angular (with full docs), and the others which are a bit hidden and just link out to GitHub READMEs? 

Essentially, it's the level of confidence we have in the integration, and the level of interest in the community for Apollo and GraphQL. For some perspective, check out this chart:

https://npm-stat.com/charts.html?package=react-apollo&package=vue-apollo&package=angular2-apollo&package=polymer-apollo&from=2015-09-01&to=2016-11-14

To more prominently feature an integration, I think we'd need:

1. Some indication that the community for that view layer is interested in GraphQL, and lots of people will use that integration
2. Someone committed to using that view layer with Apollo in a serious production app, meaning they will proactively fix bugs when they arise
3. Someone interested in updating the integration as new features arrive in Apollo core, so that people using it can have the newest stuff without waiting too long

I think (1) could be achieved by giving some talks about GraphQL in the meetups of that community and seeing what the response is like. (2) is just about finding those apps and people.

**Comment by jlevycpa** [comment]
Exactly the information I was looking for and it makes perfect sense, thanks!

Do you agree that a GitHunt port makes sense as a demo app? Have you found that this example showcases the features of Apollo well?

**Comment by stubailo** [comment]
Yes I would start with the front page app though, before getting to a complex app like githunt.

**Comment by jlevycpa** [comment]
Sample app is here: https://github.com/jlevycpa/frontpage-ember-app
